src/IceRpc.Slice.Generators/Internal/Emitter.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) ZeroC, Inc.
+
+namespace IceRpc.Slice.Generators.Internal;
+
+internal class Emitter
+{
+    internal string Emit(IReadOnlyList<ServiceClass> serviceClasses, CancellationToken cancellationToken)
+    {
+        var generatedClasses = new List<string>();
+        foreach (ServiceClass serviceClass in serviceClasses)
+        {
+            // stop if we're asked to.
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string methodModifier =
+                serviceClass.HasBaseServiceClass ? "public override" :
+                serviceClass.IsSealed ? "public" : "public virtual";
+
+            string dispatcherBlock;
+            if (serviceClass.ServiceMethods.Count > 0)
+            {
+                var dispatchBlocks = new List<string>();
+                foreach (ServiceMethod serviceMethod in serviceClass.ServiceMethods)
+                {
+                    dispatchBlocks.Add(@$"
+case ""{serviceMethod.OperationName}"":
```

Comment:
Why don't you generate a switch expression here?

Reply:
I will update the code to do so